src/amqp_management.c

```diff
@@ -999,6 +1020,14 @@ int amqp_management_close(AMQP_MANAGEMENT_HANDLE amqp_management)
     }
     else
     {
+        if (amqp_management->amqp_management_state == AMQP_MANAGEMENT_STATE_OPENING)
+        {
+            /* Codes_SRS_AMQP_MANAGEMENT_01_048: [ `amqp_management_close` on an AMQP management instance that is OPENING shall trigger the `on_amqp_management_open_complete` callback with `AMQP_MANAGEMENT_OPEN_CANCELLED`, while also passing the context passed in `amqp_management_open_async`. ]*/
+            amqp_management->on_amqp_management_open_complete(amqp_management->on_amqp_management_open_complete_context, AMQP_MANAGEMENT_OPEN_CANCELLED);
+        }
+
+        amqp_management->amqp_management_state = AMQP_MANAGEMENT_STATE_CLOSING;
+
```

Comment:
I believe this should be done before calling the open_complete. Otherwise it is very possible that in the open_complete callback another action is taken that would trigger a state change. And that would result in some undesired effects.

Reply:
Once the changes are done I'll merge it in.

---
In reply to: [174945309](https://github.com/Azure/azure-uamqp-c/pull/225#discussion_r174945309) [](ancestors = 174945309)

Reply:
Moved.